site/content/blog/2024-11-01-zap-updates-october-2024/index.md

```diff
@@ -0,0 +1,241 @@
+---
+title: "ZAP Updates - October 2024"
+summary: >
+  ZAP Updates are back after a small break.
+  Read about the updates from October, including an upgrade to Java 17, scanning of sequenced requests, a potential LLM integration, and more.
+images:
+- https://www.zaproxy.org/blog/2024-11-01-zap-updates-october-2024/images/zapbot-monthly-updates.png
+type: post
+tags:
+- blog
+- update
+date: "2024-11-01"
+authors:
+- akshath
+---
+
+It's been a while since the last monthly update, but we're back with the latest ZAP news! Here's everything that happened in October 2024.
+
+## Highlights
+
+### Java Version Upgrade
+The minimum Java version needed to build and use ZAP has been changed to Java 17.
+The weekly cross platform package, and the nightly and weekly Docker images are now using Java 17.
+Support for Java 11 will be dropped soon and the next stable release of ZAP will also use a minimum of Java 17.
+
+### Improved Passive Script Rules Performance
+Passive scripts are now cached for all passive scan threads and they are not recompiled for each thread.
+In addition, a few concurrency issues were fixed when the Graal.js engine was used with passive scripts.
+
+### Introduction of Lombok into zap-extensions
+Lombok was added into the zap-extensions repository and the developer docs were updated with references on setting it up.
+We are hoping this will help in reducing boilerplate code and make the codebase more maintainable.
+
+## Ongoing Work
+
+### Sequenced Requests
+A lot of interesting work is being done on the scanning and reporting of sequences of requests.
+This is expected to be useful for applications where the response of one request is dependent on the previous request.
+For example, in an e-commerce application, adding an item to the cart and then checking out.
+
+### LLM Integration
+A new contributor is currently working on integrating LLM APIs with ZAP.
+The idea is to allow users to enhance existing ZAP functionality and features using LLMs in a way that makes sense.
+Some examples include automatically marking alerts as false positives, or determining a sensible sequence of requests using an OpenAPI definition.
+
+### Incremental Add-on Updates
+A lot of exciting updates across add-ons are in the pipeline.
+- New GraphQL fingerprinting checks were added for a few engines
+- A new "ASCIfy" encoder was added to strip all non-ASCII characters from selected strings
+- Automation framework enhancements like simplified YAML plans, and some cool new automation jobs
+
+## Join the Discussion
+
+### ZAP Scan Policies
+
+We are considering including a default set of Active Scan Policies with all ZAP releases.
```

Comment:
Should this be worded differently, the plan is an add-on right?

Reply:
I copied that from https://groups.google.com/g/zaproxy-users/c/bEBOGB48v-I/m/3r5TRPdVAwAJ, but happy to change it. Does the following read better?

> We are considering including a default set of Active Scan Policies with ZAP as part of a new add-on.

Reply:
I think so. @psiinon ? @thc202 ?

Reply:
Yeah, that looks better.

Reply:
I'm good with that.

Reply:
Already pushed that :)